Guardfile

```diff
@@ -0,0 +1,32 @@
+guard :rspec, cmd: "bundle exec rspec" do
+  require "guard/rspec/dsl"
+  dsl = Guard::RSpec::Dsl.new(self)
+
+  # RSpec files
+  rspec = dsl.rspec
+  watch(rspec.spec_helper) { rspec.spec_dir }
+  watch(rspec.spec_support) { rspec.spec_dir }
+  watch(rspec.spec_files)
+
+  # Ruby files
+  ruby = dsl.ruby
+  dsl.watch_spec_files_for(ruby.lib_files)
+
+  # Rails files
+  rails = dsl.rails(view_extensions: %w(jbuilder))
+  dsl.watch_spec_files_for(rails.app_files)
+  dsl.watch_spec_files_for(rails.views)
+
+  watch(rails.controllers) do |m|
+    [
+      rspec.spec.("routing/#{m[1]}_routing"),
+      rspec.spec.("controllers/#{m[1]}_controller"),
+    ]
+  end
+
+  # Rails config changes
+  watch(rails.spec_helper)     { rspec.spec_dir }
+  watch(rails.routes)          { "#{rspec.spec_dir}/routing" }
+  watch(rails.app_controller)  { "#{rspec.spec_dir}/controllers" }
+
+end
```

Comment:
I'd read more on Guard and I came off with the impression that it's most useful if you've got some automated tasks like `mimification`, `preprocessing` etc in your project. This will be different with Zhishi; I wonder if we'd appreciate it knowing that it's running _probably_ redundant tests in the background as we try to come to terms with a new feature. Also given that most editors have autosave enabled, I fear for the cascade of test that might queue.

It will be interesting to have it send everyone notifications whenever boss @andela-oakinniranye makes a change to any file, I bet it'll be every 10mins.